feature/policy_forwarding/GUE/Outer_IPv4/decapsulation/otg_tests/Inner_IP/README.md

```diff
@@ -0,0 +1,337 @@
+# PF-1.4: Interface based GUE Decapsulation to IPv4 tunnel
+
+## Summary
+
+This is to test the the functionality of policy-based forwarding (PF) to decapsulate Generic UDP Encapsulation (GUE) traffic. These tests verify the use case of IPv4, and IPv6 encapsulated traffic in IPv4 GUE tuennel. The tests are meant for `Tunnel Interface` or `Policy Based` implementation of IPv4 GUE tunnel. The tests validate that the DUT performs the following action.
+
+ - Decapsulate the outer (transport) layer 3 and GUE headers of GUE packets destined to locally-configured decap VIP and matching UDP port after that it will forward them based on the exposed inner (payload) L3 header.
+ - GUE Inner protocol type must be derived from a unique DST port. If not specifically configured, then the following default DST UDP port will be used.
+    - For inner IPv4 - GUE UDP port 6080
+    - For inner IPv6 - GUE UDP port 6615
+ - Post decapsulation the DUT should copy outer TTL(and decrement) to inner header and maintain the inner DSCP vaule.
+    - If explicit configration is present to not copy the TTL, then it will be honored. 
+ - Decapsulate the packet only if it matches the locally configured GUE VIP and associated UDP port/port-range.
+    - Traffic not subject to match criteria will be forwared using traditional IP forwarding. 
+ - UDP checksum in the transport (outer) header is used for the packet integrity validation, and the DUT will ignore the GUE checksum (non-zero or all-zero). 
+
+## Testbed type
+
+* [`featureprofiles/topologies/atedut_2.testbed`](https://github.com/openconfig/featureprofiles/blob/main/topologies/atedut_2.testbed)
+
+## Procedure
+
+### Test environment setup
+
+* Create the following connections:
+* DUT has ingress and egress port connected to the ATE.
+  
+```mermaid
+graph LR; 
+A[ATE:Port1] --Ingress--> B[Port1:DUT:Port2];B --Egress--> C[Port2:ATE];
+```
+
+*  ATE Port 1: Generates GUE-encapsulated traffic with various inner (original) destinations.
+*  ATE Port 2: Receives decapsulated traffic whose inner destination matches the policy.
+  
+### DUT Configuration
+
+1.  Interfaces: Configure all DUT ports as singleton IP interfaces.
+ 
+2.  Static Routes/LSPs:
+    *  Configure an IPv4 static route to GUE decapsulation destination (DECAP-DST) to Null0.
+    *  Have policy configuration that match GUE decapsulation destination and default/non-default GUE UDP port/port-range for the decapsulation.
+       *  If udp port is not configured then the default GUE UDP port will be inherited/used.
```

Comment:
"default GUE UDP port will be inherited/used."

- Inherited from where?
- Used: Since we are using GUE v1, there isnt a GUE header that will carry details on the inner header. So looks like the ask here is for the NOS to look at the first 4 bits of the UDP payload 
    - If 0x4, then the inner header is IPv4
    - If 0x6, then the inner header is IPv6
    - I none of the above, then it is MPLS payload
 If the above is the expectation, then we should probably call it so

Reply:
I modified and mentioned the current default port (UDP port 6080 for Inner IPv4 and UDP port 6615 for Inner IPv6 traffic)

for NOS to understand which packet to perform decap and which packet to do transit, one has to specify udp destination port for the decap. Post decap, it can check first 4 bits to determine inner is ipv4 or ipv6.

I don't think GUE1 will support mpls as is because none of the above won't default to MPLS is not mentioned in draft, which is expired.